templates/tower-project.yaml

```diff
@@ -0,0 +1,338 @@
+              Action:
+                - "s3:PutObject*"
+              Condition:
+                StringEquals:
+                  s3:x-amz-acl: bucket-owner-full-control
+              Resource:
+                - !Sub "arn:aws:s3:::${TowerBucket}/*"
+            - !Ref AWS::NoValue
+          - !If
+            - HasReadWriteAccessArns
+            - Sid: GrantRemainingAccessToReadWriteAccessArns
+              Effect: Allow
+              Principal:
+                AWS: !Ref ReadWriteAccessArns
+              Action:
+                - "s3:GetObject*"
+                - "s3:DeleteObject*"
+                - "s3:*MultipartUpload*"
+                - "s3:ListBucket*"
+                - "s3:GetBucketLocation"
+              Resource:
+                - !Sub "arn:aws:s3:::${TowerBucket}"
+                - !Sub "arn:aws:s3:::${TowerBucket}/*"
+            - !Ref AWS::NoValue
+          - !If
+            - HasReadOnlyAccessArns
+            - Sid: GrantReadAccessToReadOnlyAccessArns
+              Effect: Allow
+              Principal:
+                AWS: !Ref ReadOnlyAccessArns
+              Action:
+                - "s3:ListBucket*"
+                - "s3:GetBucketLocation"
+                - "s3:GetObject"
+                - "s3:GetObjectAcl"
+              Resource:
+                - !Sub "arn:aws:s3:::${TowerBucket}"
+                - !Sub "arn:aws:s3:::${TowerBucket}/*"
+            - !Ref AWS::NoValue
+
+  TowerForgeServiceUser:
+    Type: AWS::IAM::User
+    Properties:
+      ManagedPolicyArns:
+        - !Ref TowerForgePolicyArn
+        - !Ref TowerLaunchPolicyArn
+      Policies:
+        - PolicyName: TowerForgeServiceUserS3Access
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              Effect: Allow
+              Action:
+                - "s3:ListBucket"
+                - "s3:GetObject"
+                - "s3:PutObject"
+                - "s3:PutObjectTagging"
+                - "s3:DeleteObject"
+              Resource:
+                - !Sub "arn:aws:s3:::${TowerBucket}"
+                - !Sub "arn:aws:s3:::${TowerBucket}/*"
+
+  TowerForgeServiceUserAccessKey:
+    Type: AWS::IAM::AccessKey
+    Properties:
+      UserName: !Ref TowerForgeServiceUser
+
+  TowerProjectConfigurationSecret:
+    Type: AWS::SecretsManager::Secret
+    Properties:
+      Name: !Sub "${AWS::StackName}-TowerProjectConfiguration"
+      Description: !Sub "Configuration for ${AWS::StackName} Project in Tower"
+      SecretString: !Sub
+        - >-
+          {
+            "forge_access_key_id":      "${ForgeAccessKeyId}",
+            "forge_secret_access_key":  "${ForgeSecretAccessKey}",
+            "bucket_name":              "${BucketName}",
+            "read_write_arns":          "${ReadWriteArns}",
+            "read_only_arns":           "${ReadOnlyArns}",
+            "vpc_id":                   "${VpcId}",
+            "subnet_ids":               "${SubnetIds}"
```

Comment:
these are not secrets, i usually find it to be a pain to keep non-secrets in a secrets store. is there a reason your storing them in a secrets store?

Reply:
The only reason is because I already had a secret that users need to access, so I didn't want to create yet another resource with another ARN that I need to figure out, especially how that we've moved to cross-account access, which precludes access to CloudFormation outputs.

Reply:
Another reason is that there's no way to attach a resource-based policy to an SSM Parameter for cross-account access. From what I can tell, this would require a role in our account that the external user/role would need to assume, which we just got rid of from this template.